packages/react-sdk/src/expression.ts

```diff
@@ -111,15 +111,42 @@ export const validateExpression = (
   return generateCode(expression, true, transformIdentifier);
 };
 
-export const executeExpressions = (
-  variables: Map<string, unknown>,
+const sortTopologically = (
+  list: Set<string>,
+  depsById: Map<string, Set<string>>,
+  explored = new Set<string>(),
+  sorted: string[] = []
+) => {
+  for (const id of list) {
+    if (explored.has(id)) {
+      return sorted;
```

Comment:
return looks suspicios, are you sure not continue here?
May be add test for this function?